calculator.js

```diff
@@ -5,61 +10,106 @@
  * @variable PRIVATE { Number } `total`
  * @return {object} `calculator` object that can be used
  */
-
-
+ 
+function calculatorModule() {
+      var memory = 0;
+      var total = 0;
+      var calculator = {
   /**
    * sets the `total` to the number passed in
    * @param  { Number } x
    * @return { Number }    current total
    */
-
+      load: function(x){
+          if(typeof x !== 'number'){
+            return 'Not a vaild input';
+          }
+        total = x;
+        return x;
+      },
 
   /**
    * Return the value of `total`
    * @return { Number }
    */
-
+      getTotal: function(){
+        return total;
+      },
 
   /**
    * Sums the value passed in with `total`
    * @param { Number } x
    */
-
+      add: function(x){
+        if(typeof x !== 'number'){
+            return 'Not a vaild input';
+          }
+        total += x;
+        return total;
+      },
 
   /**
    * Subtracts the value passed in from `total`
    * @param  { Number } x
    */
-
+      subtract: function(x){
+        if(typeof x !== 'number'){
+            return 'Not a vaild input';
+          }
+        total -= x;
+        return total;
+      },
 
   /**
    * Multiplies the value by `total`
    * @param  { Number } x
    */
-
+      multiply: function(x){
+        if(typeof x !== 'number'){
+            return 'Not a vaild input';
+          }
+        total *= x;
+        return total;
+      },
 
   /**
    * Divides the value passing in by `total`
    * @param  { Number } x
    */
-
+      divide: function(x){
+        if(typeof x !== 'number'){
+            return 'Not a vaild input';
+          }
+        total /= x;
+        return total;
+      },
 
   /**
    * Return the value stored at `memory`
    * @return { Number }
    */
-
+      recallMemory: function(x){
+        return memory;
+        },
 
   /**
    * Stores the value of `total` to `memory`
    */
-
-
+      saveMemory: function(){
+        memory = total;
+      },
   /**
    * Clear the value stored at `memory`
    */
+      clearMemory: function(x){
+        memory = 0;
+      },
 
   /**
    * Validation
    */
-
+      validation: function(x){
```

Comment:
So because we put the if statement in each operation for the calculator.
That is the validation. Could we make a function named validation that
would check each operation?
On Nov 1, 2015 3:54 PM, "Tony Gaskell" <notifications@github.com> wrote:

> In calculator.js
> <https://github.com/devleague/js-calculator/pull/1#discussion_r43592631>:
>
> >
> >    /**
> >     * Validation
> >     */
> > -
> > +      validation: function(x){
>
> validation method is unnecessary. I believe the comment was just a
> reminder to add validations to the previous methods. Looks good! [image:
> :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/devleague/js-calculator/pull/1/files#r43592631>.
>